src/main/java/de/tum/cit/aet/artemis/communication/repository/MessageSpecs.java

```diff
@@ -114,7 +116,9 @@ public static Specification<Post> getOwnSpecification(boolean filterToOwn, Long
                 return null;
             }
             else {
-                return criteriaBuilder.equal(root.get(Post_.AUTHOR).get(User_.ID), userId);
+                Join<Post, AnswerPost> answersJoin = root.join(Post_.ANSWERS, JoinType.LEFT);
+                Predicate searchInAnswerContent = criteriaBuilder.equal(answersJoin.get(AnswerPost_.AUTHOR).get(User_.ID), userId);
+                return criteriaBuilder.or(criteriaBuilder.equal(root.get(Post_.AUTHOR).get(User_.ID), userId), searchInAnswerContent);
```

Comment:
```suggestion
                someMeaningfulVariable = criteriaBuilder.equal(root.get(Post_.AUTHOR).get(User_.ID), userId);
                return criteriaBuilder.or(someMeaningfulVariable, searchInAnswerContent);
```
Can you extract the first criteria as well so others can understand which two criteria are used?

Reply:
Done